x-pack/metricbeat/module/azure/data_test.go

```diff
@@ -46,3 +48,113 @@ func TestManagePropertyName(t *testing.T) {
 	result = managePropertyName("Percentage CPU")
 	assert.Equal(t, result, "percentage_cpu")
 }
+
+func TestMapToKeyValuePoints(t *testing.T) {
+	timestamp := time.Now().UTC()
+	metricName := "test"
+	metricValue := 42.0
+	namespace := "test"
+	resourceId := "test"
+	resourceSubId := "test"
+	timeGrain := "PT1M"
+
+	t.Run("test aggregation types", func(t *testing.T) {
+
+		metrics := []Metric{{
+			Namespace:     namespace,
+			Names:         []string{"test"},
+			Aggregations:  "min",
+			Values:        []MetricValue{{name: metricName, min: &metricValue, timestamp: timestamp}},
+			TimeGrain:     timeGrain,
+			ResourceId:    resourceId,
+			ResourceSubId: resourceSubId,
+		}, {
+			Namespace:     namespace,
+			Names:         []string{"test"},
+			Aggregations:  "max",
+			Values:        []MetricValue{{name: metricName, max: &metricValue, timestamp: timestamp}},
+			TimeGrain:     timeGrain,
+			ResourceId:    resourceId,
+			ResourceSubId: resourceSubId,
+		}, {
+			Namespace:     namespace,
+			Names:         []string{"test"},
+			Aggregations:  "avg",
+			Values:        []MetricValue{{name: metricName, avg: &metricValue, timestamp: timestamp}},
+			TimeGrain:     timeGrain,
+			ResourceId:    resourceId,
+			ResourceSubId: resourceSubId,
+		}, {
+			Namespace:     namespace,
+			Names:         []string{"test"},
+			Aggregations:  "total",
+			Values:        []MetricValue{{name: metricName, total: &metricValue, timestamp: timestamp}},
+			TimeGrain:     timeGrain,
+			ResourceId:    resourceId,
+			ResourceSubId: resourceSubId,
+		}, {
+			Namespace:     namespace,
+			Names:         []string{"test"},
+			Aggregations:  "count",
+			Values:        []MetricValue{{name: metricName, count: &metricValue, timestamp: timestamp}},
+			TimeGrain:     timeGrain,
+			ResourceId:    resourceId,
+			ResourceSubId: resourceSubId,
+		}}
+
+		actual := mapToKeyValuePoints(metrics)
+
+		expected := []KeyValuePoint{
+			{
+				Key:           fmt.Sprintf("%s.%s", metricName, "min"),
```

Comment:
do you mind making this test a bit more realistic.

I'm thinking maybe using 2 different values for min and max. the actual average between the two values and so on.

so that we can catch something similar if it happens again.

currently all the stats for those metrics have the same value

Reply:
Yep, definitely! Adding a unique value for each aggregation.